pkg/ss/partition_mgr.go

```diff
@@ -100,15 +100,24 @@ func (m *PartitionMgr) Add(p *dstk.Partition) error {
 	return nil
 }
 
+type PartitionTask struct {
+	C chan<- error
+	Msg Msg
+}
+
 // Single threaded router. 1 channel per partition
 // path=data
-func (m *PartitionMgr) OnMsg(msg Msg) error {
+func (m *PartitionMgr) OnMsg(msg Msg) (<-chan error, error) {
 	p := m.Find(msg.Key())
+	c := make(chan error, 1)
```

Comment:
the application would like more control over the type of channel and especially the buffer size. Add a new method in base interface Msg.Channel() and let the application create it.

This also introduced another level of indirection. At runtime it causes something called pointer chasing which is believed to be one the main performance problem in java. In this case however performance is irrelevant because you are planning much expensive operations ahead, but in general try not to unecessarily nest objects